File: src/main/java/com/ocunha/domain/restaurant/filters/CustomerRatingFilterDefinition.java
package com.ocunha.domain.restaurant.filters;

import com.ocunha.domain.restaurant.model.RestaurantSearchParams;
import com.ocunha.domain.restaurant.model.model.Restaurant;
import org.springframework.stereotype.Component;

@Component
public class CustomerRatingFilterDefinition implements FilterDefinition {

    @Override
    public boolean filter(Restaurant restaurant, RestaurantSearchParams searchParams) {
        return isGreaterOrEquals(restaurant.rating(), searchParams.getRating());
    }

}

File: src/main/kotlin/com/ocunha/domain/filter/CustomerRatingFilterDefinition.kt
package com.ocunha.domain.filter

import com.ocunha.domain.restaurant.model.Restaurant
import com.ocunha.domain.restaurant.model.RestaurantSearchParams
import org.springframework.stereotype.Component

@Component
class CustomerRatingFilterDefinition : FilterDefinition {

    override fun filter(restaurant: Restaurant?, searchParams: RestaurantSearchParams?): Boolean {
        return isGreaterOrEquals(restaurant?.rating, searchParams?.rating);
    }
}